### PR TITLE
Improved Layouts

### PR DESCRIPTION
**Pull request by atomworks**
### Description

This *new feature* introduces a new layout pattern that can be used that matches up with the design systems in Figma. It is meant to handle basic responsivity, sizes and max-width simply and consistency.

* 3 New Layouts - `default`, `dashboard` and `fullscreen`
* A switch to fallback to the legacy layout (as one of the layout components was updated)
* A `UtilityHeader` to new design patterns that includes
** Breadcrumbs
** A back button
** An action button that will copy the current or custom URL to clipboard
* Allows you to send the page header, tabs and content which will be set to the new design patterns.
* A `FullWidthContentBlock` allows you to include things within your content that breaks out to the full width of the page.
* With the merging of changes to the global styles pattern for CSS, my code has been added to the new file. It uses some common variables that previously existed to handle responsiveness.
** I refactored some file locations slightly here to keep legacy files separate to make it easier to clean up later.
* `UtilityHeader` story was added and `GlobalUI` story updated to use the newer layout.

### For follow up...
* Port flexible layout components from the UI Kit experiments repo and provide an example / storybook.

 ### Considerations on Implementation
If updating to a UI kit version with this feature, if the old layouts are still being used, the alignment will break. You can simply set the prop `legacyLayout` on `GlobalUI` to correct this. The `GlobalUIPage.tsx` code in examples shows this.

 ### Reviewing/Testing steps
This can currently be seen in the examples in the new *Layouts* page.

_Storybook is pending._

## Notes For Release
This new component will provide a convenient way to layout the content in a pattern that matches that of the standardised layouts used in the newer designs. It provides a new `UtilityHeader`, already integrated; integration for the `PageHeader`; an optional dedicated tabs area to help managed tabbed content (including custom tabs); and finally the child components of this will be the main content and can be arranged in one of three views.

For layouts that have additional components that run the full width of the viewport, there is a `FullWidthContentBlock` that will allow you a simply way to break things out. If your content here needs to be aligned inside of this with the rest of the content, the responsive CSS variables that are used by the main container are available in your styled components.

**Please note,** as part of the `ContentLayouts` layouts work, some CSS had to change in the `Layouts` component of the UI kit. This now by default removes the padding to allow for ContentLayouts to manage this. For this reason, if you are updating an older project and do not wish to port to this newer pattern you will need to add the `legacyLayout` prop to the `GlobalUI` component to restore the previous padding.

With these changes there is a change to the breakpoints used. It is a slight adjustment on the largest size however please check your layouts at large sizes (> 1536) to ensure your content is scaling correctly. The UI kit now has a new default maximum width for content set at `1840px`.

### CODE REVIEW

**Review by isacoder — CHANGES_REQUESTED** [review]
Hello Leonard,

Thank you for making this great effort to refactor the layouts, It looks pretty great 😄 .

I only have two requests regarding a missing file and the sass dependency preventing the merge. Also added a few suggestions and a question regarding the height that is more of a consideration for the future than something to account in this PR.

I noticed that you have taking steps to highlight the code to be remove in the future by adding it to the legacy folder. Thanks for that too.

File: packages/ui-lib/package.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
     "react-router-dom": "^5.2.0",
     "react-scripts": "^5.0.1",
     "react-use-websocket": "^2.9.1",
+    "sass": "^1.77.8",
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/ui-lib/package.json:74]: I can't believe I missed that one...

File: packages/example/src/index.tsx [hunk 2]
```diff
@@ -13,7 +14,8 @@ import * as serviceWorker from './serviceWorker';
 import './i18n';
 import Fonts from './fonts';
 import Style from './style';
-import './theme/theme-variables.css';
+import ThemeOverwrites from './theme/ThemeOverwrites';
+import './theme/theme-variables.scss';
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/example/src/index.tsx:18]: Totally missed this one. Thank you.

File: packages/example/src/pages/TabsPage.tsx [hunk 3]
```diff
@@ -33,15 +33,15 @@ const TabsPage: React.FC = () => {
...
+          <Tab1Container>A
             <Label htmlFor='fullname' labelText='Content of tab 1' />
             <TextField fieldState='default' required label='Full Name' name='fullname' id='fullname' />
             <TextField fieldState='default' required label='Department' name='department' id='department'/>
             <Button design='primary' size='small'> Save </Button>
           </Tab1Container>
         </TabContent>
         <TabContent tabId='tab2'>
-          <Tab2Container>
+          <Tab2Container>B
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/example/src/pages/TabsPage.tsx:44]: That was just a test. Thanks for catching that.

File: packages/ui-lib/src/Global/atoms/Layout.tsx [hunk 4]
```diff
@@ -15,7 +15,8 @@ export const MobileLayout = styled.div``;
 
 
 export const Content = styled.div<{ maxWidth?: string, padBottom?: boolean }>`
-  padding: ${({padBottom = true}) => padBottom ? '70px 90px' : '70px 90px 0'};
+  /* padding: ${({padBottom = true}) => padBottom ? '70px 90px' : '70px 90px 0'}; */
+  padding: 0;
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/ui-lib/src/Global/atoms/Layout.tsx:19]: It was kept for reference but not intended for commit. Have removed!

File: packages/ui-lib/src/Layouts/atoms/FullWidthContentBlock.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+
+const FullWidthContainer = styled.div<{ $contentHeight: number }>`
+  min-height: ${({$contentHeight}) => $contentHeight + 'px'};
+`;
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/ui-lib/src/Layouts/atoms/FullWidthContentBlock.tsx:6]: This new component should size the container to the relative size of the content so despite the use of absolute positioning, the flow should stay as intended in the content.

In this instance I believe the issue with this story is that we're wrapping the story in an absolute positioned wrapper for presentation purposes that doesn't have a bottom value or height. As this container is handling the background in this case, it's stopping at the end of the content.

That said, I thought I fixed it in this story. Not sure if I am thinking of elsewhere.

I was going to try and fix this up here quickly but I couldn't see it on my local copy. Perhaps my screen contrast isn't showing it up but then I coudn't find the page where it has that Team placeholder copy either. Can you confirm which view you're looking at for this?

File: packages/ui-lib/src/Layouts/molecules/UtilityHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+  display: flex;
+  align-items: center;
+
+  > div {
+    flex: 1;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+`;
```
[the remark this answers is not in the document]
Comment (atomworks) [reply packages/ui-lib/src/Layouts/molecules/UtilityHeader.tsx:149]: Good spot. It's weird how inconsistent this is. I hadn't seen this at all until I started actively moving things about and zooming. Should be sorted now!

File: packages/ui-lib/src/Layouts/atoms/FullWidthContentBlock.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (isacoder) [reply packages/ui-lib/src/Layouts/atoms/FullWidthContentBlock.tsx:6]: Sure, the place to find it is in the Storybook, in the menu under Company there is the team view.
Screenshot 2024-07-24 at 10 41 28

[the remark this answers is not in the document]
Comment (atomworks) [reply packages/ui-lib/src/Layouts/atoms/FullWidthContentBlock.tsx:6]: Thanks! Pushed a fix for that now.

File: package-lock.json [hunk 7]
```diff
@@ -36938,6 +36944,23 @@
       "resolved": "https://registry.npmjs.org/sanitize.css/-/sanitize.css-13.0.0.tgz",
       "integrity": "sha512-ZRwKbh/eQ6w9vmTjkuG0Ioi3HBwPFce0O+v//ve+aOq1oeCy7jMV2qzzAlpsNuqpqCBjjriM1lbtZbF/Q8jVyA=="
     },
+    "node_modules/sass": {
```
Comment (isacoder) [inline package-lock.json:36947]: Please either run the npm install or checkout this file from the main branch to remove  this dependency 🙏

**Review by isacoder — APPROVED** [review]
Thanks for the fixes, all seems good just remove the changes from the package-lock.json and this branch will be ready to merge :D

File: package-lock.json (hunk 7 above)
Comment (atomworks) [reply package-lock.json]: Done!

**Pull request merged**